src/index.ts

```diff
@@ -0,0 +1,41 @@
+// @ts-ignore
+import * as S_Js from "./S_Js.bs.js";
+
+export interface Struct<Value> {
+  parse(data: any): Value;
+  parseAsync(data: any): Promise<Value>;
+  serialize(data: Value): Promise<unknown>;
+  transform<Transformed>(
+    parser: (value: Value) => Transformed
+  ): Struct<Transformed>;
+  transform<Transformed>(
+    parser: (value: Value) => Transformed,
+    serializer: (transformed: Transformed) => Value
+  ): Struct<Transformed>;
+  refine(
+    parser: (value: Value) => void,
+    serializer: (value: Value) => void
+  ): Struct<Value>;
+  asyncRefine(parser: (value: Value) => Promise<void>): Struct<Value>;
+  optional(): Struct<Value | undefined>;
+  nullable(): Struct<Value | undefined>;
+}
+
+export const string: () => Struct<string> = S_Js.string as any;
+export const bool: () => Struct<boolean> = S_Js.bool as any;
+export const int: () => Struct<number> = S_Js.$$int as any;
+export const float: () => Struct<number> = S_Js.$$float as any;
```

Comment:
Should they be called `boolean`, `integer` and `number`?

Reply:
It'd make it more friendly for Js users. Since I've already renamed `option` and `null` (because null is an invalid function name in Js), it makes sense to rename other structs too.